lib/src/license_detection/license.dart

```diff
@@ -0,0 +1,283 @@
+    final nGrams = generateChecksums(license.tokens, n);
+    final table = generateChecksumMap(nGrams);
+    return LicenseWithNGrams._(
+      nGrams,
+      table,
+      license.tokens.length < n ? license.tokens.length : n,
+      license.identifier,
+      license.content,
+      license.tokens,
+      license.occurrences,
+    );
+  }
+}
+
+/// Contains details regarding the results of corpus license match with unknown text.
+@sealed
+class LicenseMatch {
+  /// Sequence of tokens from input text that were considered a match for the detected [License].
+  @visibleForTesting
+  final List<Token> tokens;
+
+  /// [Diff]s calculated between target tokens and [license] tokens.
+  @visibleForTesting
+  final List<Diff> diffs;
+
+  /// Range of [diffs] which represents the diffs between known license and unknown license.
+  ///
+  /// Diffs lying outside of [diffRange] represent additional text in unknown license
+  /// comes before or after the detected [license](source) text.
+  @visibleForTesting
+  final Range diffRange;
+
+  /// Detected license the input have been found to match with given [confidence].
+  @visibleForTesting
+  final LicenseWithNGrams license;
+
+  /// Number of tokens claimed in this match.
+  ///
+  /// This will be updated if we come accross a match with similar
+  /// identifier to the max of number of tokens claimed of the two matches.
+  @visibleForTesting
+  int tokensClaimed;
+
+  /// Confidence score of the detected license.
+  final double confidence;
+
+  String get identifier => license.identifier;
+
+  /// Offset in the input license text considered to be possible starting point
+  /// of known license subtring.
+  int get start => tokens.first.span.start.offset;
+
+  /// Offset in the input license text considered to be possible starting point
+  /// of known license subtring.
+  int get end => tokens.last.span.start.offset;
+
+  LicenseMatch(
+    this.tokens,
+    this.confidence,
+    this.license,
+    this.diffs,
+    this.diffRange,
+  ) : tokensClaimed = tokens.length;
+}
+
+/// Generates a frequency table for the given list of [tokens].
+///
+/// [Token.value] is mapped to the number of occurrences in the license text.
+@visibleForTesting
+Map<String, int> generateFrequencyTable(List<Token> tokens) {
+  var table = <String, int>{};
+
+  for (var token in tokens) {
+    table[token.value] = table.putIfAbsent(token.value, () => 0) + 1;
+  }
+
+  return table;
+}
+
+/// Load a list of [License] from the license files in [directories].
+///
+/// The license files in [directories] are expected to be plain `.txt` with
+/// proper `utf-8` encoding. Name of the license file is expected
+/// to be in the form of [`<spdx-identifier>.txt`][1].
+/// The [directories] are not searched recursively.
+///
+/// Throws if any of the [directories] contains a file not meeting the above criteria's.
+///
+/// [1]: https://spdx.dev/ids/
+List<License> loadLicensesFromDirectories(Iterable<String> directories) {
```

Comment:
I believe this (and also `licensesFromFile`) is only used from tests. Maybe they should be moved to the test files?